### PR TITLE
Bugfix/bphh 487/password bug

### PR DESCRIPTION
**Pull request by taw27**
## Description
- Adds password validation for back-end model
- Fixes bug with password validation logic in the front-end
- Changes validation to happen on submit and shows checklist if there are errors

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x ] 🐛 Bug Fix
- [ ] 📦 Chore (Release)
- [ ] ✅ Test

## Related Tickets & Documents
https://hous-bssb.atlassian.net/browse/BPHH-487

## Steps to QA
- Go to register page
- Enter an invalid password, click submit
- Should show a checklist with errors 
Screenshot 2024-03-27 at 1 12 01 PM

### CODE REVIEW

**Comment by karenkun** [comment]
Tested it works beautifully! So awesome 👍

File: app/frontend/i18n/i18n.ts [hunk 1]
```diff
@@ -34,6 +34,15 @@ const options = {
           passwordLabel: "Password",
           forgotPassword: "Forgot password?",
           passwordTooWeak: "Password too weak",
+          passwordInvalidFormat: "Password does no meet the required format",
```
Comment (bkmaibach) [inline app/frontend/i18n/i18n.ts:37]: small typo (no -> not)

**Review by bkmaibach — APPROVED** [review]
One small typo then good to go

File: app/models/user.rb [hunk 2]
```diff
@@ -27,6 +28,8 @@ class User < ApplicationRecord
 
   enum role: { submitter: 0, review_manager: 1, reviewer: 2, super_admin: 3 }, _default: 0
 
+  validate :validate_password_complexity
```
[the remark this answers is not in the document]
Comment (bkmaibach) [reply app/models/user.rb:31]: chatgpt says its fine :D

[the remark this answers is not in the document]
Comment (taw27) [reply app/models/user.rb:31]: @bkmaibach I tested it using binding.pry

File: app/frontend/i18n/i18n.ts (hunk 1 above)
Comment (taw27) [reply app/frontend/i18n/i18n.ts:37]: good catch, will change

**Pull request merged**